Sources/NIO/CircularBuffer.swift

```diff
@@ -41,8 +63,9 @@ public struct CircularBuffer<E>: CustomStringConvertible, AppendableCollection {
     /// Allocates a buffer that can hold up to `initialRingCapacity` elements and initialise an empty ring backed by
     /// the buffer. When the ring grows to more than `initialRingCapacity` elements the buffer will be expanded.
     public init(initialRingCapacity: Int) {
-        self.bufferCapacity = initialRingCapacity
-        self.buffer = ContiguousArray<E?>(repeating: nil, count: Int(initialRingCapacity))
+        let capacity = initialRingCapacity.nextPowerOf2()
```

Comment:
do the tests really pass with this? The need a careful look if the pass as they definitely try to test the edge cases on the edge of extension

Reply:
yep they pass

Reply:
@normanmaurer then they will most likely need adjustment to still test the edge cases (unless they happen to have picked only powers of 2). Did you check the code coverage?

Reply:
nope not yet